crates/sol-macro/src/expand/contract.rs

```diff
@@ -120,64 +120,75 @@ fn expand_call_like_enum(
     assert_eq!(variants.len(), types.len());
     let name_s = name.to_string();
     let count = variants.len();
-    let min_data_len = min_data_len.min(4);
     quote! {
         #(#attrs)*
         pub enum #name {
             #(#variants(#types),)*
```

Comment:
should there be a variant for `Bare` calls, e.g. fallback/receive?

If so, that variant should contain a `Vec<u8>` containing the calldata

We can distinguish between `Bare` and `Unknown` by checking lenght. `len() % 32 == 4` is Unknown and should error. `len % 32 == 0` is `Bare`

Reply:
I think this can be done in another PR, since we don't handle any type of function (constructor, fallback...) other than literally `function`